### ISSUE TITLE
Session Upload issue. Failed to complete upload. error 

### ISSUE THREAD

**Issue by benarent** [gravitational/teleport#6094]
### Description

**What happened**:

While testing Teleport Database Access, I noticed a few issues with sessions being uploaded. This was on a db host 6.0.2 

```
WARN             Failed to complete upload. error:[
ERROR REPORT:
Original Error: *trace.BadParameterError need at least one part to complete the upload
Stack Trace:
	/go/src/github.com/gravitational/teleport/lib/events/filesessions/filestream.go:100 github.com/gravitational/teleport/lib/events/filesessions.(*Handler).CompleteUpload
	/go/src/github.com/gravitational/teleport/lib/events/stream.go:624 github.com/gravitational/teleport/lib/events.(*sliceWriter).completeStream
	/go/src/github.com/gravitational/teleport/lib/events/stream.go:497 github.com/gravitational/teleport/lib/events.(*sliceWriter).receiveAndUpload
	/opt/go/src/runtime/asm_amd64.s:1374 runtime.goexit
User Message: need at least one part to complete the upload] events/stream.go:627
WARN [DB:SERVIC] Failed to complete stream. error:[
ERROR REPORT:
Original Error: *trace.BadParameterError need at least one part to complete the upload
Stack Trace:
	/go/src/github.com/gravitational/teleport/lib/events/filesessions/filestream.go:100 github.com/gravitational/teleport/lib/events/filesessions.(*Handler).CompleteUpload
	/go/src/github.com/gravitational/teleport/lib/events/stream.go:624 github.com/gravitational/teleport/lib/events.(*sliceWriter).completeStream
	/go/src/github.com/gravitational/teleport/lib/events/stream.go:497 github.com/gravitational/teleport/lib/events.(*sliceWriter).receiveAndUpload
	/opt/go/src/runtime/asm_amd64.s:1374 runtime.goexit
User Message: need at least one part to complete the upload] events/auditwriter.go:461
```

Teleport Version 6.0.2, setup using as Database Service

**Comment by russjones** [comment]
@benarent Can you consistently reproduce this?

**Comment by jaxsong** [comment]
will this issue cause out of memory?  I met this issue on my machine

**Comment by nitrocode** [comment]
I see this error on teleport database agent 7.3.8

```
$ tsh db login --db-user=teleport --db-name=postgres snip
$ tsh db connect snip

psql: Could not connect to database:

  FATAL: PAM authentication failed for user "teleport" (SQLSTATE 28000)
```

Logs

```
2021-12-29T17:05:13Z DEBU [DB:SERVIC] Using async streamer for session e4f5a78f-9bb4-46f3-bdab-8e3813dacd51. db/streamer.go:78
2021-12-29T17:05:13Z DEBU [DB:SERVIC] Received startup message: &pgproto3.StartupMessage{ProtocolVersion:0x30000, Parameters:map[string]string{"application_name":"psql", "client_encoding":"UTF8", "database":"postgres", "user":"teleport"}}. db:snip id:e4f5a78f-9bb4-46f3-bdab-8e3813dacd51 postgres/engine.go:152
2021-12-29T17:05:13Z DEBU [RBAC]      Access to database "snip" granted, allow rule in role "db_all" matched. services/role.go:1976
2021-12-29T17:05:13Z DEBU [RBAC]      Access to database "snip" granted, allow rule in role "access" matched. services/role.go:1976
2021-12-29T17:05:13Z DEBU [DB:AUTH]   Generating RDS auth token for db[snip] identity[remote-rb-corp] dbUser[teleport] dbName[postgres]. common/auth.go:118
2021-12-29T17:05:13Z WARN             Failed to complete upload. error:[
ERROR REPORT:
Original Error: *trace.BadParameterError need at least one part to complete the upload
Stack Trace:
	/go/src/github.com/gravitational/teleport/lib/events/filesessions/filestream.go:100 github.com/gravitational/teleport/lib/events/filesessions.(*Handler).CompleteUpload
```

**Comment by zmb3** [comment]
This has been fixed with #10608
